tox.ini

```diff
@@ -1,15 +1,44 @@
 [tox]
 skipsdist = true
-envlist = py27,py36
+envlist =
+    {py27,py36,pypy}-{dj11}-{sqla11,sqla12,sqla13}
+    {py36,py37,pypy3}-{dj20,dj21,dj22}-{sqla11,sqla12,sqla13}
 
 [testenv]
 passenv =
     LC_ALL
     LANG
     HOME
     DJANGO_SETTINGS_MODULE
-whitelist_externals=make
-deps = pipenv
-commands=
-    pip install -U -r requirements.txt
+    PATH
+    LDFLAGS
+    CPPFLAGS
+basepython =
+    py27: python2.7
+    py36: python3.6
+    py37: python3.7
+    pypy: pypy
+    pypy3: pypy3
+deps =
+    dj11: django<2
+    dj20: django<2.1
+    dj21: django<2.2
+    dj22: django<2.3
+    sqla11: sqlalchemy<1.2
+    sqla12: sqlalchemy<1.3
+    sqla13: sqlalchemy<1.4
+    -rrequirements.txt
+setenv =
+    COVERAGE_FLAGS =
+whitelist_externals =
+    make
+commands =
+    pip freeze
+    make -j resetdb
     make coverage
+
+[testenv:py37-dj22-sqla13]
+setenv =
+    COVERAGE_FLAGS = --cov-fail-under=100
```

Comment:
pypy did not play nice with coverage stuff. according to local coverage report it was blank lines sometimes in docstring. could be related to docstring tests so skipping coverage minimum by default except a single py3 build where I also run lint stuff